### ISSUE TITLE
Document->download encoding issue

### ISSUE THREAD

**Issue by 0xSekar** [jlevers/selling-partner-api#231]
Testing the API with different marketplaces and noticed that there is an issue with the downloaded document for some of them.

Seems like the encoding of the downloaded report changes from market to market, for example:

- DE is ISO-8859-1
- TR is UTF-8
- JP is SJIS

The code
```
        if (!($this->contentType === ContentType::XLSX || $this->contentType === ContentType::PDF)) {
            $contents = mb_convert_encoding($contents, "UTF-8");
        }
```
seems to be unable to detect the correct origin encoding without passing from_encoding parameter as it just use the internal mbstring encoding that is UTF by default.

Tried to solve inside the API with something like:
`$contents = mb_convert_encoding($contents, "UTF-8", mb_detect_encoding());`
and even
`$contents = mb_convert_encoding($contents, "UTF-8", mb_detect_encoding(["ISO-8859-1", "UTF-8", "SJIS"]));`
But results are buggy due to the way mb_detect_encoding is implemented.

The only solution I found so far is to do something like this in my code:
```
$enc = mb_internal_encoding();
mb_internal_encoding("SJIS"); //Or any other encoding
$contents = $docToDownload->download();
mb_internal_encoding($enc);
```
Any suggestion on a better solution or a way to solve this inside the library? Will be happy to help patching the code if you can think on a better way to solve

**Comment by jlevers** [comment]
Ah, interesting. Thanks for pointing this out. I think the best solution is going to be to add an optional `$encoding` parameter to the `Document::download` method. If you want to make a PR with that, I'm happy to review it.

**Comment by jlevers** [comment]
Fixed in #233.
